monai/utils/nvtx.py

```diff
@@ -53,12 +56,46 @@ def __init__(
         name: Optional[str] = None,
         methods: Optional[Union[str, Tuple[str, ...]]] = None,
         append_method_name: Optional[bool] = None,
+        recursive: bool = False,
     ) -> None:
         self.name = name
         self.methods = methods
         self.append_method_name = append_method_name
+        self.recursive = recursive
 
     def __call__(self, obj: Any):
+        if self.recursive is True:
+            if isinstance(obj, Iterable):
+                annotated_transforms = [Range(recursive=True)(t) for t in obj]
+                return Range(self.name, recursive=False)(annotated_transforms)
+
+            from monai.transforms.compose import Compose, OneOf
+
+            if isinstance(obj, OneOf):
```

Comment:
I think line 74 and 86 could be merged using `type(obj)`:
```py
if isinstance(obj, Compose):
    ...
    Range(name)(type(obj)(...))
```

Reply:
ditto